lib/fetch/util.js

```diff
@@ -103,52 +103,57 @@ function isValidReasonPhrase (statusText) {
   return true
 }
 
-function isTokenChar (c) {
-  return !(
-    c >= 0x7f ||
-    c <= 0x20 ||
-    c === '(' ||
-    c === ')' ||
-    c === '<' ||
-    c === '>' ||
-    c === '@' ||
-    c === ',' ||
-    c === ';' ||
-    c === ':' ||
-    c === '\\' ||
-    c === '"' ||
-    c === '/' ||
-    c === '[' ||
-    c === ']' ||
-    c === '?' ||
-    c === '=' ||
-    c === '{' ||
-    c === '}'
-  )
+/**
+ * @see https://tools.ietf.org/html/rfc7230#section-3.2.6
+ * @param {number} c
+ */
+function isTokenCharCode (c) {
```

Comment:
this seems to be the only required change here (since the issue seems to be comparing the char code to the actual characters would always return false). I'd prefer if the old checks were added back and only this change made.

Reply:
@tsctx I don't believe this should be blocked by anything. The handling of pseudo-headers has nothing to do with the original issue that this fixes, right?

Reply:
@KhafraDev 
This change will fail the H2 test and should wait.

Reply:
But, I understand your thinking.

Reply:
The test isn't sound with my understanding of the fetch spec, I think the failing test(s) can be disabled for now.

Reply:
I posted more about my understanding of the fetch spec in the linked issue.

Reply:
@KhafraDev 
After the errands are completed, change the failing H2 test to be skipped.